boilerplate/src/utils/logger.js

```diff
@@ -0,0 +1,68 @@
+const elv = require('elv');
+const { createLogger, format, transports } = require('winston');
+
+const { combine, colorize, timestamp, label, printf } = format;
+require('winston-daily-rotate-file');
+const fs = require('fs');
+const path = require('path');
+
+const env = process.env.NODE_ENV;
+const logDir = 'log';
+
+// Create the log directory if it does not exist
+if (!fs.existsSync(logDir)) {
+  fs.mkdirSync(logDir);
+}
+const filename = path.join(logDir, 'results.log');
+
+// creates a new log file everyday
+// eslint-disable-next-line no-unused-vars
+const dailyRotateFileTransport = new transports.DailyRotateFile({
+  filename: `${logDir}/%DATE%-results.log`,
+  datePattern: 'YYYY-MM-DD',
+});
+
+// set the level based on env
+const level = elv.coalesce(
+  process.env.loglevel,
+  (env === 'production' || env === 'test') ? 'error' : 'debug',
+);
+
+const logger = createLogger({
+  level,
+  format: combine(
+    timestamp({
+      format: 'YYYY-MM-DD HH:mm:ss',
+    }),
+    printf(info => `${info.timestamp} ${info.level}: ${info.message}`),
+  ),
+  transports: [
+    new transports.Console({
+      level,
+      json: true,
+      stringify: obj => JSON.stringify(obj),
+      format: combine(
+        colorize(),
+        printf(
+          info => `${info.timestamp} ${info.level}: ${info.message}`,
+        ),
+      ),
+    }),
+    new transports.File({
+      filename,
+      json: true,
+      stringify: obj => JSON.stringify(obj),
+      format: combine(
+        label({ label: path.basename(process.mainModule.filename) }),
+        printf(
+          info =>
+            `${info.timestamp} ${info.level} [${info.label}]: ${info.message}`,
+        ),
+      ),
+    }),
+    // dailyRotateFileTransport
```

Comment:
What does this mean?

Reply:
Instead of writing logs to same file.. it creates a new log file everyday

Reply:
Yes, but I wasn't sure what that comment is. Is it a piece of code that was commented out, or does it mean something?

Reply:
it is dealt with in logger wrapper PR